### PR TITLE
chore: move core from root to `./packages/mockyeah`

### LINKED ISSUE #154: chore: move core from root to `packages/mockyeah`
To conform fully to monorepo architecture, so we work better with tools like `lerna` or `yarn` workspaces, let's move the code for core from root to `packages/mockyeah`. Let's do this when there are few/no PRs open or work in progress.

### PR DESCRIPTION
**Pull request by ryanricard**
resolves #154

WIP: Most everything is falling into place. There are a significant amount of linting errors we are going to have to choose whether to resolve as part of this PR or not.

Spike of orchestrating monorepo with both Lerna and Yarn Workspaces, as demonstrated here: https://medium.com/trabe/monorepo-setup-with-lerna-and-yarn-workspaces-5d747d7c0e91

Summary of changes:
- Moves root mockyeah application to `./packages/mockyeah`
- Adds yarn workspace `package.json` to root
- Adds tools local package
- Reworks linting script and config implementation
- Removes usage of gulp

### CODE REVIEW

File: packages/mockyeah/package.json [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  "main": "index.js",
+  "scripts": {
+    "start": "MOCKYEAH_ROOT=. nodemon --inspect index.js --ignore captures",
+    "test": "yarn lint && yarn test:coverage",
+    "test:watch": "yarn test:unit --watch",
+    "test:unit": "mocha test/**/*Test.js",
+    "test:coverage": "nyc yarn test:unit",
+    "test:coverage:report": "nyc report --reporter=text-lcov | coveralls",
+    "lint": "eslint -c ./eslint.config.js ./app ./lib ./server ./test",
+    "precommit": "yarn test"
```
[the remark this answers is not in the document]
Comment (ryanricard) [reply packages/mockyeah/package.json]: Husky is the bane of my existence.

**Comment by ryanricard** [comment]
It's a bit concerning that this PR's lines altered are +759 −4,918.... we should look into that happend to 4,000 lines :/

File: packages/mockyeah/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndersDJohnson) [reply packages/mockyeah/package.json]: Oh, I did not know you did not like husky. Shall we remove it?

File: packages/mockyeah-cli/package.json [hunk 2]
```diff
@@ -1,19 +1,17 @@
 {
   "name": "mockyeah-cli",
   "version": "0.2.3",
-  "description": "A cli to an invaluable service mocking platform built on Express.",
+  "description": "A powerful service mocking, recording, and playback utility.",
```
Comment (AndersDJohnson) [inline packages/mockyeah-cli/package.json]: Did you really want to change the description here?

**Comment by AndersDJohnson** [comment]
@ryanricard I did a bit more cleanup. I fixed all the lint issues in `mockyeah-cli` now that it shares rules. I also restored the `packages/mockyeah/test/fixtures` directory which was inadvertently missed due to a rule that I had to update in our root `.gitignore`.

**Pull request merged**